daemon/main.go

```diff
@@ -178,10 +187,10 @@ func getClangVersion(filePath string) (*go_version.Version, error) {
 	}
 	// at this point res is []string{"clang", "version", "maj.min.patch"}
 	verStrs := strings.Split(res[2], ".")
-	if len(verStrs) < 2 {
-		return nil, fmt.Errorf("unable to get kernel version from %q", string(verOut))
+	if len(verStrs) < 3 {
+		return nil, fmt.Errorf("unable to get clang version from %q", string(verOut))
 	}
-	return go_version.NewVersion(strings.Join(verStrs[:2], "."))
+	return go_version.NewVersion(strings.Join(verStrs[:3], "."))
```

Comment:
I realise it was like this before but what's the point here, that we verify the version has 3 components? If so, wouldn't it be shorter to write the full regex to match "clang version 1.2.3" and error if there is no match, and otherwise just return the 1.2.3 bit?

Reply:
I didn't get it, can you rephrase?